src/saleos/capacity.py

```diff
@@ -0,0 +1,499 @@
+    polarizations : int
+        Number of satellite polarizations
+
+    Returns
+    -------
+    sat_capacity : float
+        Satellite capacity in Mbps.
+
+    """
+    sat_capacity = (
+        (dl_bandwidth / 1000000) *
+        spectral_efficiency *
+        number_of_channels *
+        polarization
+    )
+
+    return sat_capacity
+
+
+def calc_constellation_capacity(channel_capacity, 
+                                number_of_channels, 
+                                polarization, 
+                                number_of_satellites):
+    """
+    Calculate the total usable constellation capacity assuming 
+    that only 50%(0.5) of constellation capacity is usable.
+
+    Constellation Capacity (Mbps) = Channel capacity (Mbps)
+                                    x Number of channels
+                                    x Polarizations
+                                    x Number of satellites
+                                    x 0.5 (50% of capacity)
+
+    Parameters
+    ----------
+    channel_capacity : float
+        The channel capacity in Mbps.
+    number_of_channels : int
+        The number of user channels per satellite.
+    polarization : int
+        The number of satellite polarizations.
+    number_of_satellites : int
+        The number of satellites.
+
+    Returns
+    -------
+    constellation_capacity : float
+        The constellation capacity in Mbps.
+
+    """
+    constellation_capacity = (channel_capacity * number_of_channels 
+                              * polarization * number_of_satellites * 0.5) 
+
+    return constellation_capacity
+
+
+def pairwise(iterable):
+    """
+    Return iterable of 2-tuples in a sliding window.
+
+    Parameters
+    ----------
+    iterable: list
+        Sliding window
+
+    Returns
+    -------
+    list of tuple
+        Iterable of 2-tuples
+
+    Example
+    -------
+    >>> list(pairwise([1,2,3,4]))
+        [(1,2),(2,3),(3,4)]
+
+    """
+    a, b = tee(iterable)
+    next(b, None)
+
+    return zip(a, b)
+
+
+def capacity_subscriber(const_cap, subscribers):
+    """
+    This function calculates usable 
+    capacity per subscriber in Mbps/subscriber.
+
+    Capacity per subscriber (Mbps/Subscriber) =
+             Constellation capacity 
+             / Number of subscribers
+
+    Parameters
+    ---------
+    const_cap : float
+        Total usable constellation capacity in Mbps.
+    subscribers : int
+        Number of subscribers.
+
+    Returns
+    -------
+    cap_sub : float
+        Capacity per subscriber in Mbps/Subscriber
+
+    """
+    cap_sub = const_cap / subscribers
+
+    return cap_sub
+
+
+def monthly_traffic(capacity_mbps):
+    """ 
+    This function calculates the monthly 
+    traffic assuming the lifespan of all 
+    constellations is 5 years and 20% 
+    accounting for traffic taking place 
+    in the busiest hour of the day.
+
+    Conversion of Mbps to monthly traffic in GB. 
+
+    Monthly traffic (GB) = (
```

Comment:
This is not filled out yet, please can you add in a much better description? I was struggling to understand it. Thanks

Reply:
Resolved